tests/Doctrine/Tests/DBAL/Schema/IndexTest.php

```diff
@@ -108,6 +108,36 @@ public function testOverrulesWithPartial()
         self::assertTrue($another->overrules($partial));
     }
 
+    /**
+     * @param string[]     $columns
+     * @param int[]|null[] $lengths1
+     * @param int[]|null[] $lengths2
+     *
+     * @dataProvider indexLengthProvider
+     */
+    public function testFulfilledWithLength(array $columns, array $lengths1, array $lengths2, bool $expected) : void
+    {
+        $index1 = new Index('index1', $columns, false, false, [], ['lengths' => $lengths1]);
+        $index2 = new Index('index2', $columns, false, false, [], ['lengths' => $lengths2]);
+
+        self::assertSame($expected, $index1->isFullfilledBy($index2));
+        self::assertSame($expected, $index2->isFullfilledBy($index1));
+    }
+
+    /**
+     * @return mixed[][]
+     */
+    public static function indexLengthProvider() : iterable
+    {
+        return [
+            'empty' => [['column'], [], [], true],
+            'same' => [['column'], [64], [64], true],
+            'different' => [['column'], [32], [64], false],
+            'sparse-different-positions' => [['column1', 'column2'], [0 => 32], [1 => 32], false],
+            'sparse-same-positions' => [['column1', 'column2'], [null, 32], [1 => 32], true],
```

Comment:
Looking at this data, are multiple `$lengths` ever possible?

Reply:
Nothing in the [documentation](https://dev.mysql.com/doc/refman/8.0/en/create-index.html) says it's impossible. If the length is omitted, the entire field length will be used. I haven't tried it myself though. Why do you think it's impossible?

Reply:
I'm wondering about this syntax: `[1 => 32, 2 => 33, 3 => 34]`

Is this something possible? I'm mostly wondering about why the type is an `array` in first place.

Reply:
Handled by https://github.com/doctrine/dbal/pull/3415#discussion_r243029973